### DISCUSSION TITLE
Best practice for replacing scheduled tasks

### DISCUSSION THREAD

**Discussion by inversiondk** [jamesmh/coravel#297] [General]
I just stumbled across Coravel by accident while looking into a better way of replacing our scheduled tasks. It looks perfect for our needs. Today we have 10+ console applications running via scheduled tasks. It would be awesome to be able to put these into one simple scheduler. 

Is there a recommended approach for replacing a number of scheduled task with Coravel? I'm a bit hesitant to create a website and then just have this schedule our tasks - because what will happen if the application pool stops (because no-one visits the website e.g.) etc.? (I think this same issue exists with Hangfire)
Would one just create a console app that runs forever on the server - or is there a preferred way of doing this? 

The product looks great. Nice and simple. I'd really like to give it a spin in regards to cleaning up our scheduled tasks.

**Comment by jamesmh** [comment]
Awesome! Hopefully Coravel works out well overall for you 🙂

And yes, there are generally two ways to run Coravel: inside a website (as you mentioned, with IIS you need to [manually configure IIS to stay alive](https://github.com/jamesmh/coravel/issues/228)). The other option is to use Coravel in a console application and register it as a Windows (or unix) service.

I wrote [an article back when .NET Core 3 was about to release.](https://www.blog.jamesmichaelhickey.com/NET-Core-Worker-Services-Background-Job-Scheduler-With-Coravel/) That should point you in the right direction.

There's also an (older) [code sample](https://github.com/jamesmh/coravel/tree/master/Samples/HostBuilderConsole) if that helps at all.

Let me know if that helps!

**Reply by inversiondk** [reply]
Thanks @jamesmh - I'll give it a look. Just wanted to make sure that I was on the right track here :)
